src/prng/isaac64.rs

```diff
@@ -312,41 +306,33 @@ fn mix(a: &mut w64, b: &mut w64, c: &mut w64, d: &mut w64,
     *h -= *d; *e ^= *g << 14; *g += *h;
 }
 
-impl Rand for Isaac64Rng {
-    fn rand<R: Rng>(other: &mut R) -> Isaac64Rng {
-        let mut key = [w(0); RAND_SIZE];
-        unsafe {
-            let ptr = key.as_mut_ptr() as *mut u8;
+impl SeedableRng for Isaac64Rng {
+    type Seed = [u8; 32];
 
-            let slice = slice::from_raw_parts_mut(ptr, RAND_SIZE * 8);
-            other.fill_bytes(slice);
+    fn from_seed(seed: Self::Seed) -> Self {
+        let mut seed_u64 = [0u64; 4];
+        le::read_u64_into(&seed, &mut seed_u64);
+        let mut seed_extended = [w(0); RAND_SIZE];
+        for (x, y) in seed_extended.iter_mut().zip(seed_u64.iter()) {
+            *x = w(*y);
         }
-        init(key, 2)
+        init(seed_extended, 2)
     }
-}
 
-impl<'a> SeedableRng<&'a [u64]> for Isaac64Rng {
-    fn reseed(&mut self, seed: &'a [u64]) {
-        *self = Self::from_seed(seed);
-    }
-
-    /// Create an ISAAC random number generator with a seed. This can
-    /// be any length, although the maximum number of elements used is
-    /// 256 and any more will be silently ignored. A generator
-    /// constructed with a given seed will generate the same sequence
-    /// of values as all other generators constructed with that seed.
-    fn from_seed(seed: &'a [u64]) -> Isaac64Rng {
-        let mut key = [w(0); RAND_SIZE];
-
-        // make the seed into [seed[0], seed[1], ..., seed[seed.len()
-        // - 1], 0, 0, ...], to fill `key`.
-        let seed_iter = seed.iter().map(|&x| x).chain(repeat(0u64));
+    fn from_rng<R: Rng>(mut rng: R) -> Result<Self, Error> {
+        // Custom `from_rng` implementations that fills the entire state
```

Comment:
Comment is incorrect; this only fills a seed

Reply:
I will change it to "Custom `from_rng` implementation that fills a seed with the same size as the entire state"